src/plugins/data_view_editor/public/components/form_fields/title_field.tsx

```diff
@@ -150,15 +160,16 @@ export const TitleField = ({
         isRollup,
         matchedIndices,
         rollupIndicesCapabilities,
-        refreshMatchedIndices,
+        // refreshMatchedIndices,
       }),
-    [isRollup, matchedIndices, rollupIndicesCapabilities, refreshMatchedIndices]
+    [isRollup, matchedIndices, rollupIndicesCapabilities] // , refreshMatchedIndices]
   );
 
   return (
     <UseField<string, IndexPatternConfig>
       path="title"
       config={fieldConfig}
+      validationDataProvider={indicesProvider}
```

Comment:
@sebelga I'm attempting to take advantage of https://github.com/elastic/kibana/pull/109238 to simplify this code. The problem I'm seeing is that the indicesProvider (and the validator) are returning the existing value, not the value thats being fetched. I can see this in the code and by updating the index pattern value - if you alternate between matching and unmatching patterns, you'll get perfectly out of sync results, one step behind.

I'm curious if you have ideas on how to address this. Thanks!